src/AbstractDifferentiation.jl

```diff
@@ -7,19 +7,34 @@ abstract type AbstractFiniteDifference <: AbstractBackend end
 abstract type AbstractForwardMode <: AbstractBackend end
 abstract type AbstractReverseMode <: AbstractBackend end
 
+"""
+    HigherOrderBackend{B}
+
+Let `ab_f` be a forward-mode automatic differentiation backend and let `ab_r` be a reverse-mode automatic differentiation backend.
+To construct a higher order backend for doing forward-over-reverse-mode automatic differentiation, use `AD.HigherOrderBackend((ab_f, ab_r))`.
+To construct a higher order backend for doing reverse-over-forward-mode automatic differentiation, use `AD.HigherOrderBackend((ab_r, ab_f))`.
+
+# Fields
+
+- `backends::B`
```

Comment:
This could be automated with DocStringExtensions.

Reply:
I would do that in a separate PR
Opening https://github.com/JuliaDiff/AbstractDifferentiation.jl/issues/107 to keep track